### DISCUSSION TITLE
Performance Problems w/Android & Umbrella (NVentive) Libraries Mobile App

### DISCUSSION THREAD

**Discussion by MttAccCanada** [unoplatform/uno#12866] [General]
Hi There,

 We are experiencing some major performance issues with Android on our cross platform app, iOS is fine but we are getting many issues.  The app is written entirely in C#; often we are seeing a lot of:

**[Choreographer] Skipped XXX Frames (many times < 200 frames)**

**XXX isn't responding** popup continuously pops up.

1) We notice this happens more consistently the longer the app is being used
2) Only happens in Android
3) We make ample use of **AsyncValuePresenter** from the NVentive Umbrella libraries and thought perhaps this could be causing the problem.

Any help at all from anyone who's experienced this can help would be great.

Thanks!!!

Dependencies:

    <PackageReference Include="GlassboxXamarin">
      <Version>6.668.2</Version>
    </PackageReference>
    <PackageReference Include="Newtonsoft.Json">
      <Version>13.0.1</Version>
    </PackageReference>
    <PackageReference Include="Portable.BouncyCastle">
      <Version>1.9.0</Version>
    </PackageReference>
    <PackageReference Include="Stetho">
      <Version>1.1.0.6382</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Licensing">
      <Version>21.1.41</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfCalendar.Android">
      <Version>20.4.0.53</Version>
    </PackageReference>
    <PackageReference Include="Syncfusion.Xamarin.SfChart.Android">
      <Version>16.1.0.26</Version>
    </PackageReference>
    <PackageReference Include="System.IdentityModel.Tokens.Jwt">
      <Version>6.15.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Analytics.AppCenter">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.App.Modules.AppCenter">
      <Version>4.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.View">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.View.Legacy">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Social.Phone">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.View.MessageDialog">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Notifications.Local">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Analytics.Core">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Environment.ApplicationLauncher">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.ActiveCommands.UWP">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.AsyncValue.UWP">
      <Version>5.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.StructuredContent.UWP">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Query">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Analyzers">
      <Version>1.6.0</Version>
    </PackageReference>
    <PackageReference Include="UmbrellaApp.Modules.Base">
      <Version>4.0.1</Version>
    </PackageReference>
    <PackageReference Include="UmbrellaApp.Modules">
      <Version>4.0.1</Version>
    </PackageReference>
    <PackageReference Include="UmbrellaApp.Modules.Views">
      <Version>4.0.1</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.App.Tasks">
      <Version>1.10.56</Version>
    </PackageReference>
    <PackageReference Include="Umbrella.Navigation">
      <Version>4.0.0</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Logging.Debug">
      <Version>1.1.1</Version>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Logging.Filter">
      <Version>1.1.1</Version>
    </PackageReference>
    <PackageReference Include="UniversalImageLoader">
      <Version>1.9.5.8565</Version>
    </PackageReference>
    <PackageReference Include="System.ValueTuple">
      <Version>4.4.0</Version>
    </PackageReference>
    <PackageReference Include="Uno.Core">
      <Version>3.0.0</Version>
    </PackageReference>
    <PackageReference Include="Uno.Serialization">
      <Version>1.25.0</Version>
    </PackageReference>
    <PackageReference Include="Uno.SourceGenerationTasks">
      <Version>4.2.0</Version>
    </PackageReference>
    <PackageReference Include="Uno.UI">
      <Version>3.10.11</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.AndroidX.AppCompat">
      <Version>1.1.0.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.AndroidX.AppCompat.AppCompatResources">
      <Version>1.1.0.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.AndroidX.DrawerLayout">
      <Version>1.1.0.1</Version>
    </PackageReference>
    <PackageReference Include="Xamarin.AndroidX.RecyclerView">
      <Version>1.1.0.1</Version>
    </PackageReference>

**Comment by francoistanguay** [comment]
Can you please send me an email at francois@platform.uno with more details about the project/problems you're getting ?
